client/gui/CAnimation.cpp

```diff
@@ -736,6 +755,13 @@ void SDLImage::verticalFlip()
 	surf = flipped;
 }
 
+// Keep the original palette, in order to do color switching operation
+void SDLImage::savePalette()
+{
+	originalPalette = new SDL_Palette();
+	memcpy(originalPalette, surf->format->palette, sizeof(SDL_Palette));

```

Comment:
Isn`t SDL has a function to copy palettes? I think `SDL_SetPaletteColors` should be used here.

Reply:
Yeah, I'm using SDL_SetPaletteColors in another function, but I didn't update here.

Reply:
Updated.